Bottom: b2be28c058bcea1f57039dd4966c64ac692e6bfa
Top:    af4cc3c64b7827f75b1392c03c0414ad0a869bca
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-25 15:04:52 -0700

rsocket: Set readfds event if rsocket has been disconnected

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/rsocket.c b/src/rsocket.c
index a3c2c6a..012bb5e 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1630,7 +1630,7 @@ rs_poll_to_select(int nfds, struct pollfd *fds, fd_set *readfds,
 	int i, cnt = 0;
 
 	for (i = 0; i < nfds; i++) {
-		if (readfds && (fds[i].revents & POLLIN)) {
+		if (readfds && (fds[i].revents & (POLLIN | POLLHUP))) {
 			FD_SET(fds[i].fd, readfds);
 			cnt++;
 		}
